### PR TITLE
Bug/849 bug with define outcome functions replicating populations

### PR DESCRIPTION
**Pull request by nanuxbe**

### CODE REVIEW

File: client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js [hunk 1]
```diff
@@ -230,6 +239,7 @@ define(['./../module', 'underscore'], function (module, _) {
           });
         }
       });
+
       console.log('currentParsetEffect', currentParsetEffect);
     }
```
Comment (jpuri) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: @kitze: these dangling console log.

File: client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js [hunk 2]
```diff
@@ -275,6 +284,7 @@ define(['./../module', 'underscore'], function (module, _) {
       $http.put('/api/project/' + vm.openProject.id + '/progsets/' + vm.selectedProgramSet.id + '/effects', vm.existingEffects).success(function (result) {
         console.log('result is', result);
         vm.existingEffects = result;
+        toastr.success('The parameters were successfully saved!', 'Success');
       });
```
Comment (jpuri) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: Toastr looks awesome @kitze

File: client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js [hunk 3]
```diff
@@ -221,7 +221,16 @@ define(['./../module', 'underscore'], function (module, _) {
...
       _.each(vm.currentParameter.populations, function(pop) {
-        var paramPops = _.filter(currentParsetEffect.parameters, {name: vm.selectedParameter.short, pop: pop.pop});
+        var paramPops = _.filter(currentParsetEffect.parameters, {name: vm.selectedParameter.short});
+        paramPops = _.filter(paramPops, function(param) {
+          if (pop.pop instanceof Array) {
+            if (param.pop.length != pop.pop.length) {
+              return false;
+            }
+            return _.difference(param.pop, pop.pop).length === 0;
+          }
```
Comment (jpuri) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: I guess a deep comparison is not needed at this point.

Comment (nanuxbe) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: I thought so as well and it was the source of the bug

Comment (jpuri) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: I would suggest to rather compare the 2 arrays on `names` or some such value.

Comment (nanuxbe) [inline client/source/js/modules/project-set/cost-coverage/cost-coverage-ctrl.js]: If you have 5 minutes let's do that over a call, it'll be faster

**Comment by jpuri** [comment]
Thanks @nanuxbe , @kitze

**Pull request merged**